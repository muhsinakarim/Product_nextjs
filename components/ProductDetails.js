// export default function ProductDetails()
const ProductDetails = (props) =>{
    return(
        <div className="relative mx-16  p-5 bg-white rounded-md box-border flex-col">
            <h2 className="mb-2 text-black font-bold font-xl">
Product No: {props.product.prodId}
            </h2>
            <p className="mb-2 font-normal text-gray">
                Product Name: {props.product.name}
            </p>
            <p className="mb-2 font-normal text-gray">
                Product Description: {props.product.description}
            </p>
            <p className="mb-3 font-normal text-gray">
                Product Price: {props.product.price}
            </p>
            <a href="#" className="absolute bottom-2 right-2 inline-flex items-center py-2 px-3 text-sm font-medium text-center bg-yellow-300 rounded-md p-5">
            ✏️
            </a>
        </div>
    );
}

export default ProductDetails