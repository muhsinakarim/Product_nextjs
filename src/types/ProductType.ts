interface ProductType{
    prodId:number;
    productName:string;
    productDescription:string;
    productPrice:number;
}
export type { ProductType};