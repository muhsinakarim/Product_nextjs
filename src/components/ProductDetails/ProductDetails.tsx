import { useEffect, useState ,FunctionComponent} from "react";
import { ProductType } from "../../types/ProductType";


const ProductDetails:FunctionComponent<ProductType> = () =>{
    const [details,setDetails]=useState<Array<ProductType>>()
    const[isLoading,setIsLoading]=useState<boolean>(false)
    useEffect(()=>{
        setIsLoading(true)
        const send = async()=>{
            const response=await fetch('http://localhost:3000/product/details');
            const responseData=await response.json();
            console.log(responseData)
            setDetails(responseData)
            setIsLoading(false)
        }
        send();
    },[])

    return(
        <div>
            {!isLoading && details && details.map((element:ProductType)=>{
                return (
                    <div className="relative w-[90%] mx-auto p-5 bg-white rounded-md box-border flex-col mb-5 mx-20 w-full">
                    <h2 className="mb-2 text-black font-bold font-xl">Product No: {element.prodId}</h2>
                    <p className="mb-2 font-normal text-gray">Product Name: {element.productName}</p>
                    <p className="mb-2  font-normal text-gray">Product Description: {element.productDescription}</p>
                    <p className="mb-3 font-normal text-gray">Product Price: {element.productPrice}</p> 
                    <a href="#" className="absolute bottom-2 right-2 inline-flex items-center py-2 px-3 text-sm font-medium text-center bg-yellow-300 rounded-md p-5">✏️</a> 
                    </div>
                )
            })}
         
          
        </div>
    );
}

export default ProductDetails