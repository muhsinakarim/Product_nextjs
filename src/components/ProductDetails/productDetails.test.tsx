import{ render } from "@testing-library/react";
import ProductDetails from "./ProductDetails";
import {ProductType} from "../../types/ProductType";

describe("Product Details component tests", ()=> {
    it("should render the UI correctly",() => {
        const mockDetails:ProductType = {
            prodId:1,
            productName:"Simpson",
            productDescription:"Toy",
            productPrice:200,
        
        };
        expect(render(<ProductDetails{...mockDetails}/>).asFragment()).toMatchSnapshot();

    });
});