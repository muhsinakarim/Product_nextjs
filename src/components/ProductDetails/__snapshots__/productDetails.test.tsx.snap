// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Details component tests should render the UI correctly 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
